csvcubed/csvcubed/models/cube/qb/components/unitscolumn.py

```diff
@@ -82,7 +80,18 @@ def validate_data(
         column_csv_title: str,
     ) -> List[ValidationError]:
         if len(self.units) > 0:
-            unique_values = {uri_safe(v) for v in set(data.unique())}
+            unique_values = set(data.unique())
+
+            map_label_to_new_uri_value = {}
+            for u in self.units:
+                if isinstance(u, NewQbUnit):
+                    map_label_to_new_uri_value.update({u.label: u.uri_safe_identifier})
```

Comment:
Why did you have to remove the set comprehension?

Reply:
pydantic, it doesent logically infer the type from an all(isinstance) conditional , leads to getting errors for QbUnit not having .uri_safe identifer.

works where you use a single condition isinstance.

i dislike the necessity tbh, but i didnt want to to alter the types.